components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/token/TokenRevocationNotifier.java

```diff
@@ -0,0 +1,39 @@
+/*
+*Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*WSO2 Inc. licenses this file to you under the Apache License,
+*Version 2.0 (the "License"); you may not use this file except
+*in compliance with the License.
+*You may obtain a copy of the License at
+*
+*http://www.apache.org/licenses/LICENSE-2.0
+*
+*Unless required by applicable law or agreed to in writing,
+*software distributed under the License is distributed on an
+*"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+*KIND, either express or implied.  See the License for the
+*specific language governing permissions and limitations
+*under the License.
+*/
+
+package org.wso2.carbon.apimgt.keymgt.token;
+
+
+/**
+ * This interface can be used to send a revoked token on realtime or to a persistent storage.
+ */
+public interface TokenRevocationNotifier {
+
+    /**
+     * Method to send the revoked token on realtime
+     * @param revokedToken revoked token to be sent
+     */
+    void sendMessageOnRealtime(String revokedToken);
```

Comment:
you need to pass the property set here right?

Reply:
<img width="1045" alt="Screen Shot 2019-04-24 at 10 27 43 AM" src="https://user-images.githubusercontent.com/23296184/56633426-afebfd80-667b-11e9-9d30-c1d7cf6ca938.png">
i am passing the notifiers when the extended class get initiated

Reply:
the problem is how does a developer who implements this interface know they have to implement a constructor with a map param?

Reply:
okay, will change accordingly.

Reply:
resolved